test/integration/world-component.spec.ts

```diff
@@ -0,0 +1,196 @@
+import { createWorldComponent } from '../../src/adapters/world'
+import { PermissionType } from '../../src/types'
+
+describe('WorldComponent', () => {
+  const mockFetch = jest.fn()
+  let worldComponent: Awaited<ReturnType<typeof createWorldComponent>>
+
+  beforeEach(async () => {
+    jest.clearAllMocks()
+
+    const mockConfig = {
+      requireString: jest.fn().mockImplementation((key) => {
+        const values = {
+          WORLD_CONTENT_URL: 'https://world-content.test',
+          LAMBDAS_URL: 'https://lambdas.test'
+        }
+        return Promise.resolve(values[key] || '')
+      }),
+      getString: jest.fn(),
+      getNumber: jest.fn(),
+      requireNumber: jest.fn()
+    }
+
+    const mockLogs = {
+      getLogger: jest.fn().mockReturnValue({
+        log: jest.fn(),
+        warn: jest.fn(),
+        info: jest.fn()
+      })
+    }
+
+    const mockCachedFetch = {
+      cache: jest.fn().mockImplementation(() => ({
+        fetch: mockFetch
+      }))
+    }
+
+    worldComponent = await createWorldComponent({
+      config: mockConfig,
+      cachedFetch: mockCachedFetch,
+      logs: mockLogs
+    })
+  })
+
+  describe('fetchWorldActionPermissions', () => {
+    it('should fetch world permissions correctly', async () => {
+      const mockPermissions = {
+        permissions: {
+          access: {
+            type: 'allow-list',
+            wallets: ['0xd9b96b5dc720fc52bede1ec3b40a930e15f70ddd']
+          },
+          deployment: {
+            type: 'allow-list',
+            wallets: []
+          },
+          streaming: {
+            type: 'allow-list',
+            wallets: ['0x123456']
+          }
+        }
+      }
+
+      mockFetch.mockResolvedValueOnce(mockPermissions)
+
+      const permissions = await worldComponent.fetchWorldActionPermissions('test-world')
+      expect(permissions).toEqual(mockPermissions.permissions)
+      expect(mockFetch).toHaveBeenCalledWith('https://world-content.test/world/test-world/permissions')
+    })
+  })
+
+  describe('hasWorldOwnerPermission', () => {
+    beforeEach(() => {
+      const mockNamesResponse = {
+        elements: [{ name: 'myworld' }, { name: 'otherworld' }]
+      }
+
+      mockFetch.mockImplementation(() => Promise.resolve(mockNamesResponse))
+    })
+
+    it('should return false when no world name is provided', async () => {
+      const result = await worldComponent.hasWorldOwnerPermission('authAddress', '')
+      expect(result).toBe(false)
+    })
+
+    it("should throw an error when world name doesn't end with .eth", async () => {
+      await expect(worldComponent.hasWorldOwnerPermission('authAddress', 'myworld')).rejects.toThrow()
```

Comment:
Checking the specific error that it throws will help us identify any potential errors when changing the function, what do you think about changing this expectation to fully check the error?
The same applies for other tests that check for errors.